action/action_deserializer_test.go

```diff
@@ -10,34 +10,50 @@ import (
 	"encoding/hex"
 	"testing"
 
+	"github.com/iotexproject/go-pkgs/hash"
 	"github.com/iotexproject/iotex-proto/golang/iotextypes"
+	. "github.com/onsi/gomega"
 	"github.com/stretchr/testify/require"
 	"google.golang.org/protobuf/proto"
 )
 
 func TestActionDeserializer(t *testing.T) {
-	r := require.New(t)
 	for _, v := range []struct {
-		id   uint32
-		hash string
+		name    string
+		chainId uint32
+		hash    string
 	}{
-		{0, "322884fb04663019be6fb461d9453827487eafdd57b4de3bd89a7d77c9bf8395"},
-		{1, "80af7840d73772d3022d8bdc46278fb755352e5e9d5f2a1f12ee7ec4f1ea98e9"},
+		{"chainID0", 0, "322884fb04663019be6fb461d9453827487eafdd57b4de3bd89a7d77c9bf8395"},
+		{"chainID1", 1, "80af7840d73772d3022d8bdc46278fb755352e5e9d5f2a1f12ee7ec4f1ea98e9"},
 	} {
-		se, err := createSealedEnvelopeWithSignature(v.id, _signByte)
-		r.NoError(err)
-		rHash, err := se.Hash()
-		r.NoError(err)
-		r.Equal(v.hash, hex.EncodeToString(rHash[:]))
-		r.Equal(v.id, se.ChainID())
-		r.Equal(_publicKey, se.SrcPubkey().HexString())
-		r.Equal(_signByte, se.Signature())
-		r.Zero(se.Encoding())
+		t.Run(v.name, func(t *testing.T) {
+			se, err := createSealedEnvelope(v.chainId)
+			NewWithT(t).Expect(err).To(BeNil())
+			NewWithT(t).Expect(se.ChainID()).To(Equal(v.chainId))
+			NewWithT(t).Expect(se.SrcPubkey().HexString()).To(Equal(_publicKey))
+			NewWithT(t).Expect(se.Signature()).To(Equal(_signByte))
+			NewWithT(t).Expect(se.Encoding()).To(BeZero())
 
-		se, err = createSealedEnvelope(v.id)
-		se1, err := (&Deserializer{}).WithChainID(true).ActionToSealedEnvelope(se.Proto())
-		r.NoError(err)
-		r.Equal(se, se1)
+			hashVal, hashErr := se.Hash()
+			_ = se.Address()
+			NewWithT(t).Expect(hex.EncodeToString(hashVal[:])).To(Equal(v.hash))
+			NewWithT(t).Expect(hashErr).To(BeNil())
+
+			se, err = createSealedEnvelope(v.chainId)
+			NewWithT(t).Expect(err).To(BeNil())
+
+			se.signature = _validSig
+			validated, err := (&Deserializer{}).WithChainID(true).
+				ActionToSealedEnvelope(se.Proto())
+			NewWithT(t).Expect(validated).NotTo(Equal(se))
+
+			NewWithT(t).Expect(se.hash).To(Equal(hash.ZeroHash256))
+			NewWithT(t).Expect(se.hashErr).To(BeNil())
+			_, _ = se.Hash()
+			_ = se.Address()
+			// after Hash() and Address() was invoked, the two value are equal
+			NewWithT(t).Expect(validated).To(Equal(se))
+		})
```

Comment:
revert these, we don't want to introduce a new testing style, pls only do minimum necessary changes to make existing test pass

Reply:
as we discussed, please revert all unnecessary changes, only need these 3 lines to make the test pass
```
_, err = se.Hash()
r.NoError(err)
se.SenderAddress()
// after Hash() and Address() was invoked, the two value are equal
```